src/pages/SearchPage/index.js

```diff
@@ -0,0 +1,181 @@
+    /* Onyx Props */
+
+    /** Beta features list */
+    betas: PropTypes.arrayOf(PropTypes.string),
+
+    /** All reports shared with the user */
+    reports: PropTypes.objectOf(reportPropTypes),
+};
+
+const defaultProps = {
+    betas: [],
+    reports: {},
+};
+
+const searchRendered = () => {
+    Timing.end(CONST.TIMING.SEARCH_RENDER);
+    Performance.markEnd(CONST.TIMING.SEARCH_RENDER);
+};
+
+function SearchPage({betas, reports}) {
+    const [isScreenTransitionEnd, setIsScreenTransitionEnd] = useState(false);
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+    const themeStyles = useThemeStyles();
+    const personalDetails = usePersonalDetails();
+
+    const offlineMessage = isOffline ? `${translate('common.youAppearToBeOffline')} ${translate('search.resultsAreLimited')}` : '';
+
+    const [searchValue, debouncedSearchValue, setSearchValue] = useDebouncedState('', 75);
+
+    useEffect(() => {
+        Timing.start(CONST.TIMING.SEARCH_RENDER);
+        Performance.markStart(CONST.TIMING.SEARCH_RENDER);
+    }, []);
+
+    const onChangeText = (text = '') => {
+        Report.searchInServer(text);
+        setSearchValue(text);
+    };
+
+    const {
+        recentReports,
+        personalDetails: localPersonalDetails,
+        userToInvite,
+        headerMessage,
+    } = useMemo(() => {
+        if (!isScreenTransitionEnd) {
+            return {
+                recentReports: {},
+                personalDetails: {},
+                userToInvite: {},
+                headerMessage: '',
+            };
+        }
+        const options = OptionsListUtils.getSearchOptions(reports, personalDetails, debouncedSearchValue.trim(), betas);
+        const header = OptionsListUtils.getHeaderMessage(options.recentReports.length + options.personalDetails.length !== 0, Boolean(options.userToInvite), debouncedSearchValue);
+        return {...options, headerMessage: header};
+    }, [debouncedSearchValue, reports, personalDetails, betas, isScreenTransitionEnd]);
+
+    const sections = useMemo(() => {
+        const newSections = [];
+        let indexOffset = 0;
+
+        if (recentReports.length > 0) {
+            newSections.push({
+                data: recentReports,
+                shouldShow: true,
+                indexOffset,
+            });
+            indexOffset += recentReports.length;
+        }
+
+        if (localPersonalDetails.length > 0) {
+            newSections.push({
+                data: localPersonalDetails,
+                shouldShow: true,
+                indexOffset,
+            });
+            indexOffset += recentReports.length;
+        }
+
+        if (userToInvite) {
+            newSections.push({
+                data: [userToInvite],
+                shouldShow: true,
+                indexOffset,
+            });
+        }
+
+        return newSections;
+    }, [localPersonalDetails, recentReports, userToInvite]);
+
+    const selectReport = (option) => {
```

Comment:
How is the performance/renders situation without these 2 being memoed in any way? Are you able to compare the two?

Reply:
@adhorodyski checked it and compared it to previous  (class based) version of this component and it's in the same place, which means 3-4 renders (depends on edge case). Also measured performance on web, though not on HTA and it looks good. If you have an access to an HT account I would gladly check againt it too!